generate_test.go

```diff
@@ -43,6 +43,56 @@ func TestFromString(t *testing.T) {
 	}
 }
 
+func TestFromStringWithServerError(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		expected := "eNpczEEKwkAMheF9TxGz7wmKXkRcxPoYho6JZEQZxLtLVHDs8vE-_mOxeTllSfQYiH5r3FGCwuWKGoPf1xify7nytNb5w27wRpDaNjwNf2Y_WzGnLXFyQBtKsTsfIrSOd_aSdfmiLt0Bc9GEIM9XAAAA__9Z7UER"
+		uri := strings.Split(r.RequestURI, "/")
+		payload := uri[len(uri)-1]
+		if payload != expected {
+			t.Errorf("FromStringWithServerError error\nexpected: %s\nactual:   %s", expected, payload)
+		}
+		imageFormat := uri[len(uri)-2]
+		if imageFormat != "jpeg" {
+			t.Errorf("FromStringWithServerError error\nexpected: %s\nactual:   %s", "jpeg", imageFormat)
+		}
+		diagramType := uri[len(uri)-3]
+		if diagramType != string(BlockDiag) {
+			t.Errorf("FromStringWithServerError error\nexpected: %s\nactual:   %s", string(BlockDiag), diagramType)
+		}
+		w.WriteHeader(400)
+		_, err := w.Write([]byte("Error 400: Unsupported output format: jpeg. Must be one of blockdiag for png, svg or pdf"))
+		if err != nil {
+			t.Errorf("error writting response:\n%+v", err)
+		}
+	}))
+	defer ts.Close()
+	port, err := strconv.ParseUint(strings.Split(ts.URL, ":")[2], 10, 16)
+	if err != nil {
+		t.Errorf("error getting the port :\n%+v", err)
+	}
+	client := New(Configuration{
+		URL:     fmt.Sprintf("http://localhost:%d", port),
+		Timeout: time.Second * 10,
+	})
+	_, err = client.FromString(`blockdiag {
+  blockdiag -> generates -> "block-diagrams";
+  blockdiag -> is -> "very easy!";
+
+  blockdiag [color = "greenyellow"];
+  "block-diagrams" [color = "pink"];
+  "very easy!" [color = "orange"];
+}`, BlockDiag, ImageFormat("jpeg"))
```

Comment:
Once https://github.com/yuzutech/kroki-go/pull/5/files merged, you could use JPEG.

Reply:
Yep good catch :+1: